web/src/SidebarPin.tsx

```diff
@@ -43,9 +49,15 @@ export const sidebarPinContext = React.createContext<SidebarPinContext>({
 export function SidebarPinContextProvider(
   props: PropsWithChildren<{ initialValue?: NonNullable<Array<string>> }>
 ) {
+  let lsc = useContext(localStorageContext)
+
   const [pinnedResources, setPinnedResources] = useState<
     NonNullable<Array<string>>
-  >(props.initialValue ?? [])
+  >(props.initialValue ?? lsc.get<Array<string>>("pinned-resources") ?? [])
```

Comment:
hmmm...i don't understand why we have two sources of truth for the initial pinned resources?

also, i think this should use a lazy initial state, as in 
https://reactjs.org/docs/hooks-reference.html#lazy-initial-state
to avoid unnecessarily reading from the localstore

Reply:
> hmmm...i don't understand why we have two sources of truth for the initial pinned resources?

`props.initialValue` is only set in tests / stories and is used to give tests / stories a way to set the initial pinned items without having to worry about storage location / format. I'll see if I can find a good way to do this via localStorageContext.

> i think this should use a lazy initial state

good call

Reply:
ya, i'd be fine with it if this was called initialValueForTesting or smth